src/mono/mono/metadata/object.c

```diff
@@ -6744,16 +6755,47 @@ mono_string_is_interned_lookup (MonoStringHandle str, gboolean insert, MonoError
 	res = (MonoString *)mono_g_hash_table_lookup (ldstr_table, MONO_HANDLE_RAW (str));
 	if (res)
 		MONO_HANDLE_ASSIGN_RAW (s, res);
-	else
+	else {
+		if (!instance_intern_table) {
+			MonoGHashTable *table = mono_g_hash_table_new_type_internal ((GHashFunc)intern_ptr_hash_internal, (GCompareFunc)intern_ptr_equal_internal, MONO_HASH_KEY_VALUE_GC, MONO_ROOT_SOURCE_DOMAIN, mono_get_root_domain (), "Domain String Pool Interned Flags");
```

Comment:
You can just pass `NULL` for the hash and comparison functions. it will then use direct pointer comparisons.

Also I think you can just say `MONO_HASH_KEY_GC` - since we know the values are exactly equal to the keys.